src/app/backing/components/builder-card/BuilderCard/BuilderCardContainer.tsx

```diff
@@ -1,39 +1,62 @@
-import { FC, useState } from 'react'
-import { useAccount } from 'wagmi'
-import { usePricesContext } from '@/shared/context/PricesContext'
+import { AllocationsContext } from '@/app/collective-rewards/allocations/context'
+import { formatSymbol } from '@/app/collective-rewards/rewards/utils/formatter'
+import { Builder } from '@/app/collective-rewards/types'
 import { RIF } from '@/lib/constants'
-import { BuilderCard, BuilderCardProps } from './BuilderCard'
+import { usePricesContext } from '@/shared/context/PricesContext'
+import { FC, useContext } from 'react'
+import { parseEther } from 'viem'
+import { useAccount } from 'wagmi'
+import { BuilderCard } from './BuilderCard'
+import { LoadingSpinner } from '@/components/LoadingSpinner'
 
-interface BuilderCardContainerProps extends BuilderCardProps {
-  onUpdateAllocation: (newAllocation: number) => void
+export interface BuilderCardContainerProps extends Builder {
+  backerRewardPercentage: NonNullable<Builder['backerRewardPercentage']>
+  estimatedRewards?: string
+  allocationTxPending?: boolean
 }
 
 export const BuilderCardContainer: FC<BuilderCardContainerProps> = ({
-  existentAllocation,
   allocationTxPending = false,
-  onUpdateAllocation,
+  address: builderAddress,
   ...props
 }) => {
   const { isConnected } = useAccount()
   const { prices } = usePricesContext()
+  const {
+    actions: { updateAllocation },
+    state: {
+      backer: { balance, amountToAllocate },
+      allocations,
+    },
+    initialState,
+  } = useContext(AllocationsContext)
   const rifPriceUsd = prices[RIF]?.price ?? 0
-  const [allocation, setAllocation] = useState<number>(existentAllocation)
+  const allocation = Number(formatSymbol(allocations[builderAddress], 'stRIF'))
+  const existentAllocation = Number(formatSymbol(initialState.allocations[builderAddress], 'stRIF'))
+
+  if (allocation === undefined || existentAllocation === undefined) {
+    return <LoadingSpinner />
+  }
```

Comment:
Are we going to show the LoadingSpinner even if the user didn't allocate anything yet?